Libplanet/Net/Messages/Pong.cs

```diff
@@ -1,13 +1,19 @@
 using System.Collections.Generic;
+using System.Numerics;
 using NetMQ;
 
 namespace Libplanet.Net.Messages
 {
     internal class Pong : Message
     {
-        public Pong(long? tipIndex)
+        public Pong()
```

Comment:
IMHO, I believe we should avoid using constructor with no parameters in this case. Because if someone uses the constructor, it must set fields (e.g., `Pong.Difficulty`, `Pong.TipIndex`), but sadly there are no tools to guarantee it at compile-time (e.g., static code analyzer). Of course, I believe the libplanet team, reviewers of this pull-request, and watchers of this project's pull-requests will fill the fields correctly. But new contributors, sometimes some of them, can have a mistake about it. And it will be critical when they weren't filled because it's the part of the network protocol.

In my thought, there are two ways to resolve the problem.

1. Avoid using a constructor with no parameters.
2. Make a static code analyzer with [Roslyn Analyzer][roslyn-analyzer] for *Libplanet*.

It's just my opinion.

[roslyn-analyzer]: https://github.com/dotnet/roslyn

Reply:
I've made this because `Pong` message is used in two ways. First case is the one which used in protocol, just for handshaking. Second case is the one used in `Swarm`, dial for preloading.

I guess adding new type of message for dial is required... Then this empty constructor will be removed naturally :)

Reply:
Okay, I hope so soon :)

Reply:
Can't agree more. Although initializing through default constructor with [object initializers][1] has been treated as the best practice for C#, the practice leads…:

1. `null` fields by mistake
2. enforcing every property/field to be nullable, even though some of them should be required business-logic-wise

I prefer to simply take all the required field values through constructor parameters with good names.

[1]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/object-and-collection-initializers#object-initializers

Reply:
> I've made this because `Pong` message is used in two ways. First case is the one which used in protocol, just for handshaking. Second case is the one used in `Swarm`, dial for preloading.
> 
> I guess adding new type of message for dial is required... Then this empty constructor will be removed naturally :)

How about opening an issue about this if there is no issue yet?

Reply:
@moreal #920 😊